gpMgmt/bin/gprebalance

```
@@ -0,0 +1,344 @@
+        parser.exit()
+
+    if options.genconf:
+        for arg in sys.argv[1:]:
+            if arg != '-g' and arg != '--gen-hosts':
+                logger.error('-g flag must be used alone')
+                parser.print_usage()
+                parser.exit()
+
+    return options, args
+
+
+def check_running_gputils(dburl: dbconn.DbURL, coordinator_data_directory: str):
+    """
+    Checks if there are any running instances of
+    gprebalance/gpbackup/gpexpand/gpshrink/gpresize
+    """
+
+    is_parallel = False
+
+    for util in ('gprebalance', 'gpexpand', 'gpshrink', 'gpresize'):
+        try:
+            with open(f'{coordinator_data_directory}/{util}.pid', 'r') as fp:
+                pid = int(fp.readline().strip())
+                if check_pid(pid):
+                    logger.error(f'{util} is already running.')
+                    sys.exit(1)
+        except IOError:
+            pass
+
+    with closing(dbconn.connect(dburl, encoding='UTF8')) as conn:
+        cursor = dbconn.query(conn, '''SELECT datid
+                    FROM pg_stat_activity
+                    WHERE application_name LIKE \'gpbackup%\' OR
+                    application_name LIKE \'gprestore%\' ''')
+        if cursor.rowcount > 0:
+            logger.error(
+                '''gpbackup/gprestore utility is already running.''')
+            sys.exit(1)
+
+    if is_gprecoverseg_running():
+        logger.error(
+            '''gprecoverseg is already running.''')
+        sys.exit(1)
+
+
+def create_pid_file(coordinator_data_directory: str):
+    """
+    Creates gprebalance pid file
+    """
+    with open(coordinator_data_directory + '/gprebalance.pid', 'w') as fp:
+        fp.write(str(os.getpid()))
+
+
+def remove_pid_file(coordinator_data_directory: str):
+    """
+    Removes gprebalance pid file
+    """
+    try:
+        os.unlink(coordinator_data_directory + '/gprebalance.pid')
+    except:
+        pass
```

I think it's a bad idea to ignore all exceptions.
```suggestion
    except IOError:
        pass
```